includes/Modules/Reader_Revenue_Manager.php

```diff
@@ -215,6 +221,78 @@ protected function create_data_request( Data_Request $data ) {
 				 */
 				$subscribewithgoogle = $this->get_service( 'subscribewithgoogle' );
 				return $subscribewithgoogle->publications->listPublications( array( 'filter' => $this->get_publication_filter() ) );
+
+			case 'POST:sync-publication-onboarding-state':
+				if ( empty( $data['publicationID'] ) ) {
+					throw new Missing_Required_Param_Exception( 'publicationID' );
+				}
+
+				if ( empty( $data['publicationOnboardingState'] ) ) {
+					throw new Missing_Required_Param_Exception( 'publicationOnboardingState' );
+				}
+
+				$return_data  = array();
+				$publications = $this->get_data( 'publications' );
+
+				if ( is_wp_error( $publications ) ) {
+					return $publications;
+				}
+
+				if ( empty( $publications ) ) {
+					return new WP_Error(
+						'publication_not_found',
+						__( 'Publication not found.', 'google-site-kit' ),
+						array( 'status' => 404 )
+					);
+				}
+
+				$publication = array_filter(
+					$publications,
+					function ( $publication ) use ( $data ) {
+						return $publication->getPublicationId() === $data['publicationID'];
+					}
+				);
+
+				if ( empty( $publication ) ) {
+					return new WP_Error(
+						'publication_not_found',
+						__( 'Publication not found.', 'google-site-kit' ),
+						array( 'status' => 404 )
+					);
+				}
+
+				$publication          = reset( $publication );
+				$new_onboarding_state = $publication->getOnboardingState();
+
+				if ( $new_onboarding_state === $data['publicationOnboardingState'] ) {
+					return function () use ( $return_data ) {
+						return (object) $return_data;
+					};
+				}
+
+				$saved_settings = false;
+				$settings       = $this->get_settings();
+
+				if ( $data['publicationID'] === $settings->get()['publicationID'] ) {
+					$settings->merge(
+						array(
+							'publicationOnboardingState' => $new_onboarding_state,
+						)
+					);
+					$saved_settings = true;
+				}
+
+				$return_data = array(
+					'publicationOnboardingState' => $new_onboarding_state,
+				);
+
+				if ( $saved_settings ) {
+					$return_data['isSavedSetting'] = true;
+				}
```

Comment:
As per my comment above, we should diverge from the IB and return `publicationID` here, while not returning `isSavedSetting`:

```suggestion
				$settings = $this->get_settings();

				if ( $data['publicationID'] === $settings->get()['publicationID'] ) {
					$settings->merge(
						array(
							'publicationOnboardingState' => $new_onboarding_state,
						)
					);
				}

				$return_data = array(
					'publicationID'              => $data['publicationID'],
					'publicationOnboardingState' => $new_onboarding_state,
				);
```

Cc @nfmohit